sched/scheduler/stateful_scheduler.go

```diff
@@ -355,39 +363,53 @@ func (s *statefulScheduler) scheduleTasks() {
 			taskId: taskId,
 			task:   taskDef,
 			nodeId: nodeId,
+
+			abortCh: make(chan interface{}, 1),
 		}
 
+		// mark the task as started in the jobState and record its taskRunner
+		jobState.taskStarted(taskId, tRunner)
+
 		s.asyncRunner.RunAsync(
-			run.run,
+			tRunner.run,
 			func(err error) {
+				log.Infof("In task callback for job: Error:%+v", err)
 				flaky := false
 				if err != nil {
 					// Get the type of error. Currently we only care to distinguish runner (ex: thrift) errors to mark flaky nodes.
 					taskErr := err.(*taskError)
+					st := err.(*taskError).st
+					aborted := (st.State == runner.ABORTED)
 					flaky = (taskErr.runnerErr != nil)
 
 					msg := "Error running job (will be retried):"
-					if taskErr.resultErr != nil && taskErr.st.State == runner.COMPLETE {
-						msg = "Error running job (quitting, tasks that run to completion are not retried):"
```

Comment:
It does look like you're right, but can you make 100% sure because I could've sworn we were retrying *all* failed tasks at some point before the introduction of sickle.